deployer/aws-lambda/content-origin-request/index.js

```diff
@@ -65,14 +76,37 @@ exports.handler = async (event, context) => {
    */
   const request = event.Records[0].cf.request;
   const host = request.headers.host[0].value.toLowerCase();
+
+  let { url, status } = resolveFundamental(request.uri);
+  if (url) {
+    return redirect(url, {
+      status,
+      cacheControlSeconds: 3600 * 24 * 30,
+    });
+  }
+
+  // Starting with /docs/ or empty path (/) should redirect to a locale.
+  // Also trim a trailing slash to avoid a double redirect.
+  if (
+    request.uri.startsWith("/docs/") ||
+    request.uri === "/" ||
+    request.uri === ""
+  ) {
+    const path = request.uri.endsWith("/")
+      ? request.uri.slice(0, -1)
+      : request.uri;
+    const language = guessLanguage(request);
+    return redirect(`/${language}${path}`);
```

Comment:
Sorry to be very particular about this ([BCP 47](https://en.wikipedia.org/wiki/IETF_language_tag)) "de" also called the language part of a locale like "de_DE" (yes "de" can also be a locale). Anyhow, I named id language because the header it uses to get it from is called ["Accept-Language"](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language). 
But I'm fine with just changing it so it's less confusing.

Reply:
Yeah, don't mind me. That word "locale" is so entrenched in MDN folklore. For example, the constant is called "VALID_LOCALES". I never know what's right. Thank you for taking the time to consider my ramblings.